### PR TITLE
feat(github): add Helm OCI support

### PR DESCRIPTION
**Pull request by dirien**
This PR add support to create OCI helm charts. No new feature is added to the existing Helm charts.

This PR continues #1768

Checklist:

* [ ] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [ ] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [ ] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [ ] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

**Comment by dirien** [comment]
CC: @sabre1041

File: .github/workflows/publish.yml
Comment (sabre1041) [inline .github/workflows/publish.yml:71]: Recommend changing to `github.actor` as it is suggested by the `login-action` [documentation](https://github.com/docker/login-action#github-container-registry)

File: .github/workflows/publish.yml
Comment (sabre1041) [inline .github/workflows/publish.yml:74]: As we are looking to publish the charts as OCI artifacts, I am wondering is there is also an interest in signing the charts using Sigstore and `cosign`. Other aspects of the Argo ecosystem are signing content and it would be good if we were able to reuse a lot of the same paradigms here.

The following is a snippet that illustrates how to not showcase how to publish the charts as OCI artifacts, but also how to sign them using Sigstore

https://github.com/sigstore/helm-charts/blob/main/.github/workflows/release.yml#L73-L83

**Review by sabre1041 — CHANGES_REQUESTED** [review]
@dirien Added a few comments and suggestions

File: .github/workflows/publish.yml
Comment (dirien) [reply .github/workflows/publish.yml:74]: I thought about the same but wanted to create a separate PR for  it! For now this PR is to get OCI charts in place

Comment (sabre1041) [reply .github/workflows/publish.yml:74]: Would we want to start with unsigned charts and then move to signed charts or tackle both at the same time. I defer to the project maintainers and community on the thoughts of the collective group

File: .github/workflows/publish.yml
Comment (dirien) [reply .github/workflows/publish.yml]: Done! @sabre1041 thanks for spotting this!

File: .github/workflows/publish.yml
Comment (dirien) [reply .github/workflows/publish.yml:74]: I am fine with booth! If the maintainers want it added into this PR happy to do so.

**Comment by dirien** [comment]
Thanks @sabre1041 for the comments. Appreciate your feedback! Let's get this done and we finally have OCI charts for Argo! 🥳

File: .github/workflows/publish.yml
Comment (mkilchhofer) [reply .github/workflows/publish.yml:74]: I think we can skip cosign for now. We should clarify with  argoproj maintaines what key to use for this.

But can we use the codeblock from cosign mentioned above? Your PR have two issues
- `find` needs to filter for tgz only as we also generate the provenance file for the old non-OCI index
- `/charts` doesn't exists as a repo in github.com/argoproj

Comment (dirien) [reply .github/workflows/publish.yml:74]: Hey @mkilchhofer,

- Done for filtering only the *.tgz! Thanks for the hint!

- /charts will be created and is part of the name. The finale OCI artefact name would look like this e.g: `oci://ghcr.io/argo-helm/charts/argo-cd`

Comment (mkilchhofer) [reply .github/workflows/publish.yml:74]: I think the first element after
ghci.io would be `argoproj` then.

I prefer having `ghcr.io/argoproj/argo-helm` for consitency:

```
helm push "${pkg}" oci://ghcr.io/${GITHUB_REPOSITORY}
```

Comment (dirien) [reply .github/workflows/publish.yml:74]: Sounds good! Changed it.

**Review by mkilchhofer — APPROVED** [review]
Lgtm. Thanks for implementing this @dirien !!🥳

**Comment by sabre1041** [comment]
Glad to see this get pushed through. thanks again @dirien

**Comment by mkilchhofer** [comment]
Wohoooo, first chart arrived via OCI ❤️ :
https://github.com/argoproj/argo-helm/pkgs/container/argo-helm%2Fargo-cd

thanks again @dirien

**Comment by sabre1041** [comment]
@dirien @mkilchhofer was thinking about the same. I would wait until we have official documentation on consumption. I would also like to continue the Sigstore discussion and can help lead that if desired on the signing requirements for the charts.

**Comment by dirien** [comment]
@sabre1041 What about meeting next year and discuss the next steps regarding sigstore integration?

**Comment by sabre1041** [comment]
> @sabre1041 What about meeting next year and discuss the next steps regarding sigstore integration?

Certainly up for the discussion. While I do advocate for the use for sigstore, lets have a discussion on methods for the incorporation of additional signing methods

**Pull request merged**
